starknet_py/net/account/account_test.py

```diff
@@ -12,8 +13,18 @@
 
 @pytest.mark.asyncio
 @pytest.mark.parametrize("net", (TESTNET, TESTNET2, MAINNET))
-async def test_get_balance_default_token_address(net):
-    client = GatewayClient(net=net)
+@pytest.mark.parametrize(
+    "call_contract",
+    [
+        "starknet_py.net.gateway_client.GatewayClient.call_contract",
+        "starknet_py.net.full_node_client.FullNodeClient.call_contract",
+    ],
+)
+async def test_get_balance_default_token_address(net, call_contract):
+    if "gateway" in call_contract:
+        client = GatewayClient(net=net)
+    else:
+        client = FullNodeClient(node_url=net + "/rpc")
```

Comment:
I don't really like `if else` like this in tests but alternative would be duplication I think.

Wdyt @drknzz @war-in

Reply:
I think it's the best we can get

Reply:
How about that?

```
client = (
        GatewayClient(net=net)
        if "gateway" in call_contract
        else FullNodeClient(net=net + "/rpc")
    )
```

I know it is still `if/else` but in a nicer format